dev/travis-before-install.sh

```diff
@@ -0,0 +1,34 @@
+set -e
+date
+sudo apt-get update -qq
+sudo apt-get install -qq build-essential pv autoconf automake libtool curl make \
+   g++ unzip libboost-dev libboost-test-dev libboost-program-options-dev \
+   libevent-dev automake libtool flex bison pkg-config g++ libssl-dev xmlstarlet
+date
+mkdir protobuf_install
+pushd protobuf_install
+wget https://github.com/google/protobuf/archive/v3.5.1.tar.gz -O protobuf-3.5.1.tar.gz
+tar xzf protobuf-3.5.1.tar.gz
+cd protobuf-3.5.1
+./autogen.sh
+./configure
+make
+sudo make install
+sudo ldconfig
+protoc --version
+popd
+date
+pwd
+wget -nv http://archive.apache.org/dist/thrift/0.9.3/thrift-0.9.3.tar.gz
+tar zxf thrift-0.9.3.tar.gz
+cd thrift-0.9.3
+chmod +x ./configure
+./configure --disable-gen-erl --disable-gen-hs --without-ruby --without-haskell --without-erlang --without-php --without-nodejs
+sudo make install
+cd ..
+branch_specific_script="dev/travis-before-install-${TRAVIS_BRANCH}.sh"
+if [[ -e "$branch_specific_script" ]]
```

Comment:
Should this go before the protobuf and thrift install since it doesn't require those steps?

Reply:
I also considered that option, but feature branches may need those, depending on what they want to do (for example, build parquet-format). Since in general I expect this check to fail very seldom, I think generally it is not a waste of resources to build protobuf and thrift first.

Reply:
Sounds fine to me, although we can always change it later when we have more checks.

Reply:
We already know that the bloom-filter branch wants to build its own parquet-format (https://github.com/apache/parquet-mr/pull/587/files). I would like their .travis.yml modification to be moved to the branch-specific script instead, and for that we need to invoke the script after building thrift.